app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -504,4 +313,69 @@ public void onServiceDisconnected(ComponentName arg0) {
             mBound = false;
         }
     };
+
+    private String findSmsData(String[] prebuffer,String[] postbuffer){
```

Comment:
The same loop range 4 times. Could you write this simpler? It's too complex and it's not efficient at all.